test/unit/rand/src/RandTest.cpp

```diff
@@ -47,34 +48,27 @@ BOOST_AUTO_TEST_SUITE(rand_)
 //#############################################################################
 class RandTestKernel
 {
-public:
-#if BOOST_COMP_GNUC
-    #pragma GCC diagnostic push
-    #pragma GCC diagnostic ignored "-Wunused-variable"
-#endif
-    //-----------------------------------------------------------------------------
     ALPAKA_NO_HOST_ACC_WARNING
     template<
-        typename TAcc>
-    ALPAKA_FN_ACC auto operator()(
-        TAcc const & acc) const
-    -> void
+        typename TAcc,
+        typename T_Generator
+    >
+    ALPAKA_FN_ACC void
+    genNumbers(
+        TAcc const & acc,
+        T_Generator & gen
+    ) const
     {
-        auto gen(alpaka::rand::generator::createDefault(acc, 12345u, 6789u));
-
-#if BOOST_COMP_CLANG
-    #pragma clang diagnostic push
-    #pragma clang diagnostic ignored "-Wunused-variable"
-#endif
 // gcc 5.4 in combination with nvcc 8.0 fails to compile the CPU STL distributions when --expt-relaxed-constexpr is enabled
 // /usr/include/c++/5/cmath(362): error: calling a __host__ function("__builtin_logl") from a __device__ function("std::log") is not allowed
-#if !((BOOST_COMP_GNUC >= BOOST_VERSION_NUMBER(5, 0, 0)) && (BOOST_COMP_NVCC >= BOOST_VERSION_NUMBER(8, 0, 0)) && (BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(9, 0, 0)))
+#if !((BOOST_COMP_GNUC >= BOOST_VERSION_NUMBER(5, 0, 0)) && (BOOST_COMP_NVCC == BOOST_VERSION_NUMBER(8, 0, 0)))
```

Comment:
@BenjaminW3 this was removed with #474

Since this branch still support CUDA 7.X, should we keep this? Or is this a legacy NVCC 9 change only?

Reply:
I am not sure. As long as the CI is happy I do not have a problem to change a test on the old branch.
You made the check more strict so that it should not apply anymore because there is not exact nvcc 8.0.0. Previously all 8.*.* have been filtered out.

Reply:
hm, actually there are two CUDA 8 point releases. Not sure if even in `dev` a `8.0.0` would be catched...
https://gist.github.com/ax3l/9489132

Reply:
Will change it to the same as in #590.

Reply:
Ok, looks to me that the 0.3.X branch has the usual CMake checks from #566 already in place:
https://github.com/ComputationalRadiationPhysics/alpaka/blob/0.3.2/alpakaConfig.cmake#L382-L414

So we can apply #590 and remove the pre-compiler guards here as well